### DISCUSSION TITLE
Install Rustdesk Pro on Rocky Linux

### DISCUSSION THREAD

**Discussion by Caracal-HacK** [rustdesk/rustdesk-server-pro#160] [Q&A] [answered]
Hello everyone,

I recently chose to use the rustdesk pro product, after launching the installation on Rocky Linux, everything went smoothly until the NGINX part.

I wanted to use the script available on the official Rustdesk doc and I couldn't, because some Linux commands are not adapted to the Rocky Linux distribution.

As a first step, can you include a second script in your development?

Here's my situation:

I've edited the script and adapted the commands to the distribution I've chosen.

Install dependencies = OK
Creation of firewall rules = OK
Creation of Rustdesk directories = OK
Install executables in /usr/bin : OK
Download and extract tar.gz = OK
Create hbbs and hbbr services = OK

The problem is on the nginx part, the rustdesk documentation notes that a directory named "sites-available" and "sites-enabled" exists in the nginx folder, but this directory doesn't exist. So I can't launch the RustDesk administration console. Can you please help me so that I can use the chosen product.

-----------------------------------------------------------------------------------------------------------------------------------------

Bonjour à tous,

J'ai récemment choisi d'utiliser le produit rustdesk pro, après avoir lancé l'installation sur Rocky Linux, tout c'est passé jusqu'à la partie NGINX.

J'ai voulu utiliser le script disponible sur la doc officielle Rustdesk et je n'ai pas pu, car certaines commandes Linux ne sont pas adaptées à la distribution Rocky Linux.

Dans un premier temps, pouvez-vous prévoir un second script dans votre développement ?

Je vous expose ma situation :

J'ai donc édité le script et adapté les commandes à la distribution que j'ai choisie.

Installation des dépendances = OK
Création des règles firewall = OK
Création des répertoires Rustdesk = OK
Installation des exécutables dans /usr/bin : OK
Téléchargement et extraction tar.gz = OK
Création des services hbbs et hbbr = OK

La ou sa coince c'est sur la partie nginx, il est noté dans la documentation rustdesk qu'un répertoire nommé "sites-available" et "sites-enabled" existe dans le dossier nginx, hors ce répertoire n'existe pas. Je ne peux donc pas lancer la console d'administration RustDesk. Pouvez-vous s'il vous plaît m'aider afin que je puisse utiliser le produit choisi.

**Comment by rustdesk** [comment]
https://stackoverflow.com/questions/17413526/nginx-missing-sites-available-directory

![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/739ee82f-82b2-4b72-acab-3d3fb61b9205)
![image](https://github.com/rustdesk/rustdesk-server-pro/assets/71636191/fb2bad88-9455-49d6-beb1-5b65eec61449)

**Reply by rustdesk** [reply]
We will update the script to take this into account.

**Reply by rustdesk** [reply]
@21pages please help me fix this and test on rocky linux.

https://github.com/rustdesk/rustdesk-server-pro/blob/29daee0608830d5fb691a4edecc30416e574666e/install.sh#L404

**Comment by rustdesk** [comment, marked as answer]
Please try out https://github.com/rustdesk/rustdesk-server-pro/pull/161

**Comment by Caracal-HacK** [comment]
Hi, I've managed to configure the nginx side of the rustdesk.conf file. Isn't the location of the rustdesk working directory missing? How can nginx know where rustdesk points if it is not mentioned : 

```
server {
  server_name ${RUSTDESK_DOMAIN};
      location / {
           proxy_set_header X-Real-IP \$remote_addr;
           proxy_set_header X-Forwarded-For \$proxy_add_x_forwarded_for;
        proxy_pass http://127.0.0.1:21114/;
      }
}
```
Isn't the "location" part of your script incomplete?

Thank you

**Reply by rustdesk** [reply]
` proxy_pass http://127.0.0.1:21114/;` told nginx. It does not have to be a working directory.

**Comment by Caracal-HacK** [comment]
Thanks for your feedback, I tried to install rustdesk server pro on Windows server with the executable provided to see the differences in use. And do you know why the application launches and closes so early? 
The rustdesk server pro application on Windows does not open after installation. When I install it on a Windows 10 machine, it works perfectly. Is Rustdesk compatible with Windows Server?

**Reply by rustdesk** [reply]
If that does not work, please try out https://rustdesk.com/docs/en/self-host/rustdesk-server-oss/windows/

**Reply by rustdesk** [reply]
https://www.reddit.com/r/rustdesk/comments/187tnkj/error_on_setting_up_server_on_windows_server_2012/
